recipes/ncbi-cxx-toolkit-public/all/conanfile.py

```diff
@@ -0,0 +1,294 @@
+                    break
+
+        _allcomponents = self._parse_option(self.options.with_components)
+        _required_components.update(_allcomponents)
+
+        if len(_required_components) > 0:
+            _todo = _required_components.copy()
+            _required_components.clear()
+            _next = set()
+            while len(_todo) > 0:
+                for _component in _todo:
+                    if not _component in _required_components:
+                        _required_components.add(_component)
+                        if _component in self._tk_dependencies["dependencies"].keys():
+                            for _n in self._tk_dependencies["dependencies"][_component]:
+                                if not _n in _required_components:
+                                    _next.add(_n)
+                _todo = _next.copy()
+                _next.clear()
+
+        if len(_required_components) == 0:
+            _required_components.update( self._tk_dependencies["components"])
+        else:
+            for component in _required_components:
+                self._componenttargets.update(self._tk_dependencies["libraries"][component])
+
+        requirements = set()
+        for component in _required_components:
+            libraries = self._tk_dependencies["libraries"][component]
+            for lib in libraries:
+                if lib in self._tk_dependencies["requirements"].keys():
+                    requirements.update(self._tk_dependencies["requirements"][lib])
+
+        for req in requirements:
+            pkgs = self._translate_req(req)
+            if pkgs is not None:
+                for pkg in pkgs:
+                    print("Package requires ", pkg)
+                    self.requires(pkg)
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        if self.settings.os not in ["Linux", "Macos", "Windows"]:
+            raise ConanInvalidConfiguration("This operating system is not supported")
+        if is_msvc(self):
+            check_min_vs(self, 192)
+            if self.options.shared and is_msvc_static_runtime(self):
+                raise ConanInvalidConfiguration("This configuration is not supported")
+        else:
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(f"This version of {self.settings.compiler} is not supported")
+        if cross_building(self):
+            raise ConanInvalidConfiguration("Cross compilation is not supported")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        apply_conandata_patches(self)
+        root = os.path.join(os.getcwd(), "CMakeLists.txt")
+        with open(root, "w", encoding="utf-8") as f:
```

Comment:
This could just be a file that is checked into git, you do not need to dynamically write this

Reply:
As a matter of fact, "root" CMakeLists.txt is located in [src](https://github.com/ncbi/ncbi-cxx-toolkit-public/blob/master/src/CMakeLists.txt) directory.
Why not in the tree root? There is no definite answer to this question. Tradition, maybe even convenience... 
At present there is no consensus whether it makes sense to have it in the root.
It is required to create the Conan package, so here it is.